.buildkite/scripts/build_and_upload_images.sh

```diff
@@ -41,22 +41,35 @@ make build-image-prerequisites
 # Build targets
 docker buildx bake --file="${BUILDX_BAKE_FILE}" --progress "plain" "${BUILDX_TARGET}"
 
-# Cloudsmith may be having trouble with `buildx --push`, so, experimenting with
-# uploading each one individually.
-{
-    mapfile -t fully_qualified_images < <(
-        docker buildx bake \
-            --file="${BUILDX_BAKE_FILE}" \
-            --print "${BUILDX_TARGET}" |
-            jq --raw-output '.target | to_entries | .[] | .value.tags[0]'
-    )
-    for fq_image in "${fully_qualified_images[@]}"; do
-        echo "--- Pushing ${fq_image}"
-        retry 3 \
-            docker push "${fq_image}"
-    done
+push() {
+    # https://github.com/grapl-security/issue-tracker/issues/931
+    # Try the `buildx --push` first; if retried on Buildkite, do the slower
+    # manual approach.
+
+    if [[ "${BUILDKITE_RETRY_COUNT:-0}" -eq 0 ]]; then
+        retry 2 \
+            docker buildx bake --file="${BUILDX_BAKE_FILE}" \
+            --progress "plain" \
+            --push \
+            "${BUILDX_TARGET}"
+    else
+        # Upload each image in serial, not parallel
+        mapfile -t fully_qualified_images < <(
+            docker buildx bake \
+                --file="${BUILDX_BAKE_FILE}" \
+                --print "${BUILDX_TARGET}" |
+                jq --raw-output '.target | to_entries | .[] | .value.tags[0]'
+        )
+        for fq_image in "${fully_qualified_images[@]}"; do
+            echo "--- Pushing ${fq_image}"
+            retry 3 \
```

Comment:
I'd probably also remove this retry, but not do any sort of custom exit code logic.